src/System.Private.ServiceModel/tests/Scenarios/Security/TransportSecurity/Tcp/StreamingTests.4.1.0.cs

```diff
@@ -223,6 +223,81 @@ public static void NetTcp_TransportSecurity_Streamed_RoundTrips_String()
                nameof(Ambient_Credentials_Available))]
     [Issue(832, Framework = FrameworkID.NetNative)] // Windows Stream Security is not supported in NET Native
     [OuterLoop]
+    public static void NetTcp_TransportSecurity_Streamed_MultipleReads()
+    {
+#if FULLXUNIT_NOTSUPPORTED
+        bool root_Certificate_Installed = Root_Certificate_Installed();
+        bool client_Certificate_Installed = Client_Certificate_Installed();
+        bool windows_Authentication_Available = Windows_Authentication_Available();
+        bool ambient_Credentials_Available = Ambient_Credentials_Available();
+
+        if (!root_Certificate_Installed ||
+            !client_Certificate_Installed ||
+            !windows_Authentication_Available ||
+            !ambient_Credentials_Available)
+        {
+            Console.WriteLine("---- Test SKIPPED --------------");
+            Console.WriteLine("Attempting to run the test in ToF, a ConditionalFact evaluated as FALSE.");
+            Console.WriteLine("Root_Certificate_Installed evaluated as {0}", root_Certificate_Installed);
+            Console.WriteLine("Client_Certificate_Installed evaluated as {0}", client_Certificate_Installed);
+            Console.WriteLine("Windows_Authentication_Available evaluated as {0}", windows_Authentication_Available);
+            Console.WriteLine("Ambient_Credentials_Available evaluated as {0}", ambient_Credentials_Available);
+            return;
+        }
+#endif
+        string testString = new string('a', 20000);
+        NetTcpBinding binding = null;
+        ChannelFactory<IWcfService> factory = null;
+        IWcfService serviceProxy = null;
+        Stream stream = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            binding = new NetTcpBinding(SecurityMode.Transport);
+            binding.TransferMode = TransferMode.Streamed;
+            factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.Tcp_Transport_Security_Streamed_Address));
+            serviceProxy = factory.CreateChannel();
+            stream = StringToStream(testString);
+
+            // *** EXECUTE *** \\
+            var returnStream = serviceProxy.EchoStream(stream);
+            var ms = new MemoryStream((int)stream.Length);
+            var buffer = new byte[10];
```

Comment:
>10 [](start = 34, length = 2)

Just curious: did you choose a tiny buffer just to hammer the ReadAsync calls?  Is there any advantage to using a number 20000 is not divisible by to verify what happens on a partial read of < buffer.Length?